core/utils/histogram_test.cc

```diff
@@ -34,38 +34,48 @@
 namespace {
 
 TEST(HistogramTest, U32Quartiles) {
+  // 1002 is out of range, thus will be floored to 1000
+  const std::vector<uint32_t> values = {1, 2, 3, 4, 5, 1002};
+
   Histogram<uint32_t> hist(1000, 1);
-  std::vector<uint32_t> values = {1, 2, 3, 4, 5, 1001};
   for (uint32_t x : values) {
-    hist.insert(x);
+    hist.Insert(x);
   }
-  ASSERT_EQ(1, hist.above_threshold());
-  ASSERT_EQ(2, hist.min());
-  ASSERT_EQ(6, hist.max());
-  ASSERT_EQ(4, hist.avg());
-  ASSERT_EQ(5, hist.count());
-  ASSERT_EQ(20, hist.total());
-  ASSERT_EQ(1, hist.percentile(25));   // 25th percentile
-  ASSERT_EQ(2, hist.percentile(50));   // 50th percentile
-  ASSERT_EQ(3, hist.percentile(75));   // 75th percentile
-  ASSERT_EQ(5, hist.percentile(100));  // 100th percentile
+
+  auto ret = hist.Summarize({25.0, 50.0, 75.0, 100.0});
+
+  EXPECT_EQ(1, ret.above_range);
+  EXPECT_EQ(1, ret.min);
+  EXPECT_EQ(1000, ret.max);
+  EXPECT_EQ(169, ret.avg);
+  EXPECT_EQ(6, ret.count);
+  EXPECT_EQ(1015, ret.total);
+  EXPECT_EQ(2, ret.percentile_values[0]);     // 25th percentile
+  EXPECT_EQ(4, ret.percentile_values[1]);     // 50th percentile
```

Comment:
Not sure why this test only fails with an actual value of 3 for some of the builds. When I trace the new `Summarize()` by hand I also get 3

Reply:
Yeah I am curious too. Perhaps it's because of `-Ofast` option we give to g++. That optimization option includes some non-standard floating point stuff.

One hack would be using 49.9999 or 50.0001 instead 😆